spec/spec.md

```diff
@@ -1013,62 +1012,197 @@ in activities performed by entities participating in the message thread.
-to a Hub User's non-public data.
+to various data and functionality provided by an Identity Hub. Permissions employ a 
+capabilities-based architecture that allows for DID-based authorization and delegation 
+of authorized capabilities to others, if allowed by the Identity Hub owner.
 
 #### Request
 
-```json
-{ // Message
-  "data": {...},
-  "descriptor": { // Message Descriptor
-    "method": "PermissionsRequest",
-    "schema": "https://schema.org/MusicPlaylist"
-  }
-}
-```
+`PermissionsRequest` messages are JSON objects that include general [Message Descriptor](#message-descriptors) properties and the following additional properties, which ****must**** be composed as follows:
 
-#### Query
+- The message object ****MUST**** contain a `descriptor` property, and its value ****MUST**** be a JSON object composed as follows:
+  - The object ****MUST**** contain a `method` property, and its value ****MUST**** be the string `PermissionsRequest`.
+  - The object ****MUST**** contain an `objectId` property, and its value ****MUST**** be an [[spec:rfc4122]] UUID Version 4 string representing the reply object.
+  - The object ****MUST**** contain a `requester` property, and its value ****MUST**** be the DID URI of the party requesting the permission.
+  - The object ****MAY**** contain a `description` property, and its value ****MUST**** be a string that the requesting party uses to communicate what the permission is being used for.
+  - The object ****MUST**** contain a `ability` property, and its value ****Must**** be an object of the following properties:
+    - The object ****MUST**** contain a `can` property, and its value ****Must**** be an object of the following properties:
+      - The object ****MUST**** contain a `method` property, and its value ****MUST**** be the interface method the requesting party wants to invoke.
+      - The object ****MAY**** contain a `schema` property, and its value ****Must**** be a URI string that indicates the schema of the associated data.
+    - The object ****MAY**** contain a `conditions` property, and its value ****Must**** be an object of the following properties:
+      - The object ****MAY**** contain a `attestation` property. If this property is not present it reflects the same directive as the `0` value below. If the property is present its value ****Must**** be one of the following enums representing different data signing directives:
+        - `0` - indicates that the object ****MUST NOT**** be signed.
+        - `1` - indicates that the object ****MUST**** be signed using a key linked to the DID of the Hub owner or authoring party, represented in the [[spec:rfc7515]] JSON Web Signature (JWS) format.
+      - The object ****MAY**** contain an `encryption` property. If this property is not present it reflects the same directive as the `0` value below. If the property is present its value ****Must**** be one of the following enums representing different data encryption directives:
```

Comment:
change to `boolean` because absence of `encryption` implies that the data may or may not be encrypted. If this the case, the sender would then have no idea how to encrypt the data was provided alongside a permission

Reply:
There's a good reason to keep it an integer, which I will detail in the PR update.